pallets/order-book/src/lib.rs

```diff
@@ -664,13 +650,36 @@ pub mod pallet {
 				partial_fulfillment,
 				currency_in: order.asset_in_id,
 				currency_out: order.asset_out_id,
-				fulfillment_amount: buy_amount,
-				sell_rate_limit: order.max_sell_rate,
+				fulfillment_amount: amount_out,
+				price,
 			});
 
 			Ok(())
 		}
 
+		pub fn market_price(
+			currency_in: T::AssetCurrencyId,
+			currency_out: T::AssetCurrencyId,
+		) -> Result<T::SellRatio, DispatchError> {
+			let price = T::PriceProvider::get(
+				&T::MarketFeederId::get(),
+				&(currency_in, currency_out).into(),
+			)?;
+
+			Ok(match price {
+				Some(price) => price,
+				None => {
+					let price = T::PriceProvider::get(
+						&T::MarketFeederId::get(),
+						&(currency_out, currency_in).into(),
+					)?
+					.ok_or(Error::<T>::MarketPriceNotFound)?;
+
+					T::SellRatio::one().ensure_div(price)?
+				}
+			})
```

Comment:
Nit: Could we determine final price outside the returning `Ok` scope? Seems preferable to me to `unwrap_or` the above price which is `Option<T::SellRatio>`.

Reply:
Actually, that was my first attempt, doing it with `unwrap_or_else(|| ..)` to avoid extra computation in case we do not take that branch. 

Nevertheless, because the inner closure would return an error, it became messy to get the value correctly.